server/server.js

```diff
@@ -153,7 +153,14 @@ app.post(`${apiBaseUrl}/approve-user`,
   keycloak.allowRolesMiddleware('ministry_of_health'),
   asyncMiddleware(async (req, res) => {
     await validate(AccessRequestApproval, req.body);
-    await keycloak.approvePendingRequest(req.body.userId, req.body.role, [req.body.region]);
+    const promises = req.body.sites.map((siteId) => dbClient.db[collections.EMPLOYER_SITES].findDoc({ siteId }));
+    const results = await Promise.all(promises);
+    const siteRegions = results.map(item => item[0].healthAuthority);
+    const userRegions = siteRegions.includes(req.body.region) ?
+      siteRegions
+      :
+      [...siteRegions, req.body.region];
```

Comment:
Okay I see what we're doing here. This is essentially a dedupe for `siteRegions` + `req.body.region`. Alternative here:
```
const joinedRegions = [...siteRegions, req.body.region];
const uniqueRegions = [...new Set(joinedRegions)];
```

Reply:
Yeah this is what I had to state at for a little bit. Adding everything then casting to set is how I would usually approach this as well, if we both find that more readable maybe it's worth swapping to it